velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -298,13 +297,55 @@ class SharedArbitrationTest : public exec::test::HiveConnectorTestBase {
 
   static inline FakeMemoryOperatorFactory* fakeOperatorFactory_;
   std::unique_ptr<memory::MemoryManager> memoryManager_;
-  SharedArbitrator* arbitrator_;
+  SharedArbitrator* arbitrator_{nullptr};
   RowTypePtr rowType_;
   VectorFuzzer::Options fuzzerOpts_;
   RowVectorPtr vector_;
   std::atomic_uint64_t numAddedPools_{0};
 };
 
+namespace {
+std::unique_ptr<folly::Executor> newMultiThreadedExecutor() {
+  return std::make_unique<folly::CPUThreadPoolExecutor>(32);
+}
+
+struct TestParam {
+  bool isSingleThreaded{false};
+};
+} // namespace
+
+/// A test fixture that runs cases within multi-threaded execution mode.
+class SharedArbitrationTest : public SharedArbitrationTestBase {
+ protected:
+  void SetUp() override {
+    SharedArbitrationTestBase::SetUp();
+    executor_ = newMultiThreadedExecutor();
+  }
+};
+/// A test fixture that runs cases within both single-threaded and
+/// multi-threaded execution modes.
+class SharedArbitrationTestWithThreadingModes
+    : public testing::WithParamInterface<TestParam>,
+      public SharedArbitrationTestBase {
```

Comment:
Why we can't mark SharedArbitrationTest a parameterized gtest? Thanks!

Reply:
Would you mean to run all the test cases in this file with both execution modes?

That's on my plan though I suggest we do the changes smoothly by separating PRs so keep them small. The current enabled 3 test cases are most important ones so should have higher priority,

Reply:
Then, we can just leave this test refactor in a separate PR? thanks1

Reply:
Separated to https://github.com/facebookincubator/velox/pull/10730